app/src/main/java/cl/maleb/testcarsales/ui/covid/CovidFragment.kt

```diff
@@ -0,0 +1,112 @@
+package cl.maleb.testcarsales.ui.covid
+
+import android.app.DatePickerDialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import androidx.lifecycle.Observer
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import cl.maleb.testcarsales.data.Result
+import cl.maleb.testcarsales.databinding.CovidLayoutBinding
+import cl.maleb.testcarsales.di.Injectable
+import cl.maleb.testcarsales.di.injectViewModel
+import java.util.*
+import javax.inject.Inject
+
+class CovidFragment : Fragment(), Injectable {
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    private lateinit var viewModel: CovidViewModel
+    private lateinit var binding: CovidLayoutBinding
+
+    // Calendar variables
+    val c = Calendar.getInstance()
+    val year = c.get(Calendar.YEAR)
+    val month = c.get(Calendar.MONTH)
+    val day = c.get(Calendar.DAY_OF_MONTH)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        viewModel = injectViewModel(viewModelFactory)
+
+        // inflate layout
+        binding = CovidLayoutBinding.inflate(inflater, container, false)
+        binding.btnDateSelection.setOnClickListener {
+            // dialog
+            val dpd = DatePickerDialog(
+                requireContext(),
+                DatePickerDialog.OnDateSetListener { view, year, monthOfYear, dayOfMonth ->
+                    viewModel.Date =
+                        viewModel.parseDateFromCalendar(dayOfMonth, monthOfYear + 1, year)
+
+                    viewModel.loadCovid().observe(viewLifecycleOwner, Observer {
```

Comment:
This has to be improved. A new observer is being created each time the btnDateSelection is clicked

Reply:
The viewmodel needs to have a way to load new data but not each time create a new livedata object

Reply:
Here the thing should be like this: 

OnDateSetListener {
    viewModel.fetch(date)
}

And below where you have an observer, subscribe to the changes and update your layout

Reply:
Now this has improved. In CovidViewModel I added a transformations var with switchmap on it, that every time a fetch a date or in this case my new var "dateLiveData" changes its value, the method will call the API with the date.